### PR TITLE
Start using configcat feature flags instead of env in service portal

### PR DESCRIPTION
**Pull request by dabbeg**
# \<Description\>

I finished implementing a way to use environment variables for static frontends here #2972
I have updated this PR to use that functionality. 

## What

Integrate Configcat into service portal instead of using environment variables

## Why

Because Configcat is our newly decided feature flagging solution

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/service-portal/src/hooks/useRoutes/useRoutes.ts [hunk 1]
```diff
@@ -3,12 +3,14 @@ import { useEffect } from 'react'
...
 import { useModuleProps } from '../useModuleProps/useModuleProps'
+import { useModules } from '../useModules/useModules'
 import { User } from 'oidc-client'
 import { ApolloClient, NormalizedCacheObject } from '@apollo/client'
 import flatten from 'lodash/flatten'
 
 export const useRoutes = () => {
-  const [{ modules }, dispatch] = useStore()
+  const [_, dispatch] = useStore()
+  const { modules } = useModules()
```
Comment (darrikonn) [inline apps/service-portal/src/hooks/useRoutes/useRoutes.ts:13]: In the file above you fetch the modules from the store, using the `useStore` hook.
Which one do we want to use?

File: apps/service-portal/src/hooks/useModules/useModules.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import { userManager } from '../../utils/userManager'
+import { featureFlaggedModules, ModuleKeys } from '../../store/modules'
+import { environment } from '../../environments'
+
+const featureFlagClient = createClient({
+  sdkKey: environment.featureFlagSdkKey,
+})
+
+export const useModules = () => {
+  const [{ modules }, dispatch] = useStore()
```
Comment (darrikonn) [inline apps/service-portal/src/hooks/useModules/useModules.ts:18]: ^ here

File: apps/service-portal/src/hooks/useRoutes/useRoutes.ts (hunk 1 above)
Comment (darrikonn) [reply apps/service-portal/src/hooks/useRoutes/useRoutes.ts:13]: Ahh I see, it's in the `useModules` hook itself.

On closer inspection, I see that you're following the way it's done for the other hooks. So you can ignore my comment :)

**Review by darrikonn — APPROVED** [review]
LGTM

File: apps/service-portal/src/hooks/useRoutes/useRoutes.ts (hunk 1 above)
Comment (dabbeg) [reply apps/service-portal/src/hooks/useRoutes/useRoutes.ts]: Good catch I switched this out to use the store instead of the useModules hook

**Review by darrikonn — DISMISSED** [review]

**Review by Thorrik58 — DISMISSED** [review]
LGTM

**Review by Thorrik58 — APPROVED** [review]

**Review by darrikonn — APPROVED** [review]

**Pull request merged**